libraries/AP_Scripting/applets/mount-viewpro-driver.lua

```diff
@@ -0,0 +1,372 @@
+      end
+      break
+    end
+
+    -- waiting for header4
+    if parse_state == PARSE_STATE_WAITING_FOR_HEADER4 then
+      if b == HEADER4 then
+        parse_state = PARSE_STATE_WAITING_FOR_DATA
+      else
+        -- unexpected byte so reset parsing state
+        parse_state = PARSE_STATE_WAITING_FOR_HEADER1
+      end
+      break
+    end
+
+    -- waiting for data
+    if parse_state == PARSE_STATE_WAITING_FOR_DATA then
+
+      -- check for crc
+      if parse_data_bytes_recv >= 54 then
+        if b == parse_expected_crc then
+          -- complete parsing
+          local roll_deg = bytearray_to_angle_deg(parse_data_buff[5], parse_data_buff[6], parse_data_buff[7], parse_data_buff[8])
+          local pitch_deg = bytearray_to_angle_deg(parse_data_buff[23], parse_data_buff[24], parse_data_buff[25], parse_data_buff[26])
+          local yaw_deg = bytearray_to_angle_deg(parse_data_buff[41], parse_data_buff[42], parse_data_buff[43], parse_data_buff[44])          
+          mount:set_attitude_euler(MOUNT_INSTANCE, roll_deg, pitch_deg, yaw_deg)
+       end
+       parse_state = PARSE_STATE_WAITING_FOR_HEADER1
+       break
+      end
+
+      -- add latest byte to crc and buffer
+      parse_expected_crc = (parse_expected_crc + b) & 0xFF
+      parse_data_bytes_recv = parse_data_bytes_recv + 1
+      parse_data_buff[parse_data_bytes_recv] = b
+    end
+  end
+end
+
+-- write a byte to the uart and update the checksum
+function write_byte(b, checksum)
+  if b == nil or checksum == nil then
+    gcs:send_text(3, "ViewPro: failed to write byte") -- MAV_SEVERITY_ERR
+    return
+  end
+  local byte_to_write = b & 0xFF
+  uart:write(byte_to_write)
+
+  return (checksum + byte_to_write) & 0xFF
+end
+
+-- get lowbyte of a number
+function lowbyte(num)
+  return num & 0xFF
+end
+
+-- get highbyte of a number
+function highbyte(num)
+  return (num >> 8) & 0xFF
+end
+
+-- convert 4 byte array to angle in degrees1
+function bytearray_to_angle_deg(b0, b1, b2, b3)
+  return ((b3 << 24) + (b2 << 16) + (b1 << 8) + b0) / ANGLE_DEG_TO_OUTPUT_SCALING
+end
+
+-- write num to the uart and update the checksum
+function write_number(num, checksum)
+  if not num or not checksum then
+    gcs:send_text(3, "ViewPro: failed to write number") -- MAV_SEVERITY_ERR
+    return
+  end
+  checksum = write_byte(lowbyte(num), checksum)
+  checksum = write_byte(highbyte(num), checksum)
+  return checksum
+end
+
+-- convert angle (in degres) to output scaling
+function angle_to_output_scaling(angle_deg)
+  local scaled_angle = math.floor(angle_deg * ANGLE_DEG_TO_OUTPUT_SCALING + 0.5)
+  if scaled_angle > 8192 then
+    scaled_angle = 8192
+  end
+  if scaled_angle < -8192 then
+    scaled_angle = -8192
+  end
+  return scaled_angle
+end
+
+-- convert rate (in deg/sec) to output scaling
+function rate_to_output_scaling(rate_degs)
+  local scaled_rate = math.floor(rate_degs / 0.122 + 0.5)
+  if scaled_rate > 8192 then
+    scaled_rate = 8192
```

Comment:
```suggestion
  if scaled_rate > 8191 then
    scaled_rate = 8191
```
Usually it wraps at 8191

Reply:
txs again for the review.  In this case though, I think 180deg (which is a valid target) becomes 8192 so I will leave it as-is for now anyway.